diff --git a/node_modules/@astrojs/markdoc/components/Renderer.astro b/node_modules/@astrojs/markdoc/components/Renderer.astro
index c26d92a..0f9b0eb 100644
--- a/node_modules/@astrojs/markdoc/components/Renderer.astro
+++ b/node_modules/@astrojs/markdoc/components/Renderer.astro
@@ -3,6 +3,8 @@
 import type { Config } from '@markdoc/markdoc';
 import Markdoc from '@markdoc/markdoc';
 import { ComponentNode, createTreeNode } from './TreeNode.js';
+import MarkdocExtendedParser from '@nvanexan/markdoc-extensions/dist/index.mjs';
+import { extendedNodes } from '@nvanexan/markdoc-extensions';
 
 type Props = {
 	config: Config;
@@ -11,8 +13,16 @@ type Props = {
 
 const { stringifiedAst, config } = Astro.props as Props;
 
+config.nodes = {
+ ...config.nodes,
+  ...extendedNodes,
+};
+
 const ast = Markdoc.Ast.fromJSON(stringifiedAst);
-const content = await Markdoc.transform(ast, config);
+
+const parser = new MarkdocExtendedParser({footnotes: true, highlights: true});
+const result = parser.parse(ast);
+const content = await Markdoc.transform(result, config);
 ---
 
 {
