diff --git a/node_modules/@astrojs/markdoc/components/Renderer.astro b/node_modules/@astrojs/markdoc/components/Renderer.astro
index 2706040..95120d3 100644
--- a/node_modules/@astrojs/markdoc/components/Renderer.astro
+++ b/node_modules/@astrojs/markdoc/components/Renderer.astro
@@ -3,6 +3,8 @@
 import type { Config, RenderableTreeNodes } from '@markdoc/markdoc';
 import Markdoc from '@markdoc/markdoc';
 import { ComponentNode, createTreeNode } from './TreeNode.js';
+import MarkdocExtendedParser from '@nvanexan/markdoc-extensions/dist/index.mjs';
+import { extendedNodes } from '@nvanexan/markdoc-extensions';
 
 type Props = {
 	config: Config;
@@ -11,12 +13,19 @@ type Props = {
 
 const { stringifiedAst, config } = Astro.props as Props;
 
+const updatedConfig = {
+	...config,
+	nodes: {
+		...config.nodes,
+		...extendedNodes,
+	},
+};
+
 const ast = Markdoc.Ast.fromJSON(stringifiedAst);
-// The AST may be an array, and `transform` has overloads for arrays and non-array cases,
-// However TypeScript seems to struggle to combine both overloads into a single signature.
-// Also, `transform` returns a promise here but the types don't reflect that.
-// @ts-expect-error
-const content = (await Markdoc.transform(ast, config)) as RenderableTreeNodes;
+
+const parser = new MarkdocExtendedParser();
+const parsedAst = parser.parse(ast as any);
+const content = (await Markdoc.transform(parsedAst, updatedConfig)) as RenderableTreeNodes;
 const treeNode = await createTreeNode(content);
 ---
 
